dipy/viz/ui.py

```diff
@@ -1292,3 +1292,249 @@ def key_press(i_ren, obj, textbox_object):
+        disk.SetOuterRadius(outer_radius)
+        disk.SetRadialResolution(10)
+        disk.SetCircumferentialResolution(50)
+        disk.Update()
+
+        # Mapper
+        mapper = vtk.vtkPolyDataMapper2D()
+        mapper.SetInputConnection(disk.GetOutputPort())
+
+        # Actor
+        self.slider_disk = vtk.vtkActor2D()
+        self.slider_disk.SetMapper(mapper)
+
+        self.slider_disk.SetPosition(self.center[0], self.center[1])
+
+        # /Slider Disk
+
+        # Slider Text
+        self.text = TextActor2D()
+
+        self.text.position = (self.left_x_position-50, self.center[1]-10)
+        percentage = self.calculate_percentage(current_val=int(self.current_state))
+        self.text.message = percentage
+        self.text.font_size = text_size
+        # /Slider Text
+
+    def get_actors(self):
+        """ Returns the actors that compose this UI component. 
+        
+        """
+        return [self.slider_line, self.slider_disk, self.text.get_actor()]
+
+    def set_position(self, position):
+        """ Sets the disk's position.
+        
+        Parameters
+        ----------
+        position : (float, float)
+            The absolute position of the disk (x, y).
+        
+        """
+        x_position = position[0]
+        if x_position < self.center[0] - self.length/2:
+            x_position = self.center[0] - self.length/2
+        if x_position > self.center[0] + self.length/2:
+            x_position = self.center[0] + self.length/2
+        self.slider_disk.SetPosition(x_position, self.center[1])
+        self.current_state = x_position
+
+    def calculate_percentage(self, current_val):
+        """ Calculates the percentage to be displayed.
+        
+        Parameters
+        ----------
+        current_val : int
+            Absolute value of the disk's x position.
+        
+        """
+        percentage = int(((current_val-self.left_x_position)*100)/(self.right_x_position-self.left_x_position))
+        if percentage < 0:
+            percentage = 0
+        if percentage > 100:
+            percentage = 100
+        return str(percentage) + "%"
+
+    def set_percentage(self, current_val):
+        """ Sets text percentage.
+        
+        Parameters
+        ----------
+        current_val : int
+            This is the x-position of the slider in the 2D coordinate space
+            and not the percentage on the base scale.
+            
+        """
+        self.current_state = current_val
+        percentage = self.calculate_percentage(current_val=current_val)
+        self.text.message = percentage
+
+    def set_center(self, position):
+        """ Sets the center of the slider to position.
+        
+        Parameters
+        ----------
+        position : (float, float)
+            The new center of the whole slider (x, y).
+        
+        """
+        self.slider_line.SetPosition(position[0] - self.length / 2, position[1] - self.line_width / 2)
```

Comment:
Max character per line should be 80 chars. Run `pep8 ui.py` and `flakes8 ui.py`, nothing should output to the console. if the code respects the PEP8 standard.